src/universal/redux/authDuck.js

```diff
@@ -31,58 +32,72 @@ export default function reducer(state = initialState, action = {}) {
   }
 }
 
-export function setAuthToken(authToken, metaProfile = {}) {
-  const profile = metaProfile || {};
-  if (!authToken) {
-    throw new Error('setAuthToken action created with undefined authToken');
-  }
-  let obj = null;
-  try {
-    obj = jwtDecode(authToken);
-  } catch (e) {
-    throw new Error(`unable to decode jwt: ${e}`);
-  }
-  cashay.create({httpTransport: new ActionHTTPTransport(authToken)});
-  if (typeof __PRODUCTION__ !== 'undefined' && __PRODUCTION__) {
-    /*
-     * Sentry error reporting meta-information. Raven object is set via SSR.
-     * See server/Html.js for how this is initialized
-     */
-    Raven.setUserContext({ // eslint-disable-line no-undef
-      id: obj.sub,
-      email: profile.email
-    });
-  }
-  return {
-    type: SET_AUTH_TOKEN,
-    payload: {
-      obj,
-      token: authToken
-    },
-    meta: {
-      analytics: {
-        eventType: EventTypes.identify,
-        eventPayload: {
-          userId: obj.sub,
-          traits: {
-            avatar: profile.picture,
-            createdAt: profile.created_at,
-            email: profile.email,
-            name: profile.name
+export function setAuthToken(authToken, newProfile) {
+  return (dispatch, getState) => {
+    const cachedProfile = selectProfile(getState());
+    const profile = newProfile || cachedProfile;
+
+    if (newProfile) {
+      // update cached profile for everybody:
+      dispatch(setProfile(newProfile));
+    }
+
+    if (!authToken) {
+      throw new Error('setAuthToken action created with undefined authToken');
+    }
+    let obj = null;
+    try {
+      obj = jwtDecode(authToken);
+    } catch (e) {
+      throw new Error(`unable to decode jwt: ${e}`);
+    }
+
+    cashay.create({httpTransport: new ActionHTTPTransport(authToken)});
+
+    if (typeof __PRODUCTION__ !== 'undefined' && __PRODUCTION__) {
+      /*
+       * Sentry error reporting meta-information. Raven object is set via SSR.
+       * See server/Html.js for how this is initialized
+       */
+      Raven.setUserContext({ // eslint-disable-line no-undef
```

Comment:
I kinda wanna use raven-js for the clientside instead of it magically appearing in the global scope. That means no CDN (or rather using our own CDN). Thoughts?